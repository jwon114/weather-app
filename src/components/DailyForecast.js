import React from 'react'
import PropTypes from 'prop-types';
import Date from '../utils/date';

const DailyForecast = (props) => (
  <div className="daily-forecast">
    <div className="daily-forecast__day-container">
      <span className="daily-forecast__day">{Date.getDayOfWeek(props.day)}</span>
      <span className="daily-forecast__temperature">{Math.floor(props.temperature)}&deg;</span>
    </div>
    <div className="daily-forecast__forecast-container">
      <img className="daily-forecast__icon-code" src={`http://openweathermap.org/img/wn/${props.iconCode}.png`} alt="weather forecast icon" />
      <span className="daily-forecast__forecast">{props.forecast}</span>
    </div>
  </div>
);

DailyForecast.propTypes = {
  day: PropTypes.number,
  temperature: PropTypes.number,
  iconCode: PropTypes.string,
  forecast: PropTypes.string
};

export default DailyForecast;