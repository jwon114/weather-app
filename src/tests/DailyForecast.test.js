import React from 'react';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';
import DailyForecast from '../components/DailyForecast';

test('should render Forecast correctly', () => {
  const root = create(<DailyForecast />);
  
  expect(root.toJSON()).toMatchSnapshot();
});

test('should display the Day, Temperature, Weather Description', () => {
  const testData = [
    {
      dt: 1587985200, // Monday
      temp: { day: 18.69 },
      weather: [{ description: 'clear sky', icon: '01d' }]
    }
  ];

  const { getByAltText, getByText } = render(<DailyForecast 
                                                key={testData[0].dt}
                                                day={testData[0].dt} 
                                                temperature={testData[0].temp.day}
                                                iconCode={testData[0].weather[0].icon}
                                                forecast={testData[0].weather[0].description} />);
  
  expect(getByText(/mon/i)).toBeInTheDocument();
  expect(getByText((content, node) => node.textContent === 18 + 'Â°')).toBeInTheDocument();
  expect(getByText(/clear sky/i)).toBeInTheDocument();
  expect(getByAltText(/weather forecast icon/i).src).toEqual(`https://openweathermap.org/img/wn/${testData[0].weather[0].icon}.png`);
});