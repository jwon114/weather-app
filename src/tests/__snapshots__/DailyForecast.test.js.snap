// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Forecast correctly 1`] = `
<div
  className="daily-forecast"
>
  <div
    className="daily-forecast__day-container"
  >
    <span
      className="daily-forecast__day"
    />
    <span
      className="daily-forecast__temperature"
    >
      NaN
      Â°
    </span>
  </div>
  <div
    className="daily-forecast__forecast-container"
  >
    <img
      alt="weather forecast icon"
      className="daily-forecast__icon-code"
      src="http://openweathermap.org/img/wn/undefined.png"
    />
    <span
      className="daily-forecast__forecast"
    />
  </div>
</div>
`;
